x/concentrated-liquidity/lp_test.go

```diff
@@ -551,6 +552,148 @@ func (s *KeeperTestSuite) TestisInitialPositionForPool() {
+			expectedError:         true,
+		},
+		"try updating with ticks outside existing position's tick range": {
+			poolId:                    1,
+			ownerIndex:                0,
+			lowerTick:                 DefaultUpperTick + 1,
+			upperTick:                 DefaultUpperTick + 100,
+			liquidityDelta:            DefaultLiquidityAmt,
+			amount0Expected:           sdk.NewInt(18395),
+			amount1Expected:           sdk.ZeroInt(),
+			expectedPositionLiquidity: DefaultLiquidityAmt,
+			// only liquidity that has been added from create position.
+			// no liquidity has been added by updating position since tick was outside range.
+			expectedPoolLiquidity: DefaultLiquidityAmt,
+			expectedError:         false,
+		},
+		"error: invalid pool id": {
+			poolId:          2,
+			ownerIndex:      0,
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			liquidityDelta:  DefaultLiquidityAmt,
+			amount0Expected: DefaultAmt0Expected,
+			amount1Expected: DefaultAmt1Expected,
+			expectedError:   true,
+		},
+		"new position when calling update position": {
+			poolId:                    1,
+			ownerIndex:                1, // using a different address makes this a new position
+			lowerTick:                 DefaultLowerTick,
+			upperTick:                 DefaultUpperTick,
+			liquidityDelta:            DefaultLiquidityAmt,
+			amount0Expected:           DefaultAmt0Expected,
+			amount1Expected:           DefaultAmt1Expected,
+			expectedPositionLiquidity: DefaultLiquidityAmt,
+			expectedPoolLiquidity:     DefaultLiquidityAmt.Mul(sdk.NewDec(2)),
+			expectedError:             false,
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			tc := tc
+			s.SetupTest()
+
+			// create a CL pool
+			s.PrepareConcentratedPool()
+
+			// create position
+			// Fund test account and create the desired position
+			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin(ETH, DefaultAmt0), sdk.NewCoin(USDC, DefaultAmt1)))
+			_, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(
+				s.Ctx,
+				1,
+				s.TestAccs[0],
+				DefaultAmt0, DefaultAmt1,
+				sdk.ZeroInt(), sdk.ZeroInt(),
+				DefaultLowerTick, DefaultUpperTick,
+			)
+			s.Require().NoError(err)
+
+			// system under test
+			actualAmount0, actualAmount1, err := s.App.ConcentratedLiquidityKeeper.UpdatePosition(
+				s.Ctx,
+				tc.poolId,
+				s.TestAccs[tc.ownerIndex],
+				tc.lowerTick,
+				tc.upperTick,
+				tc.liquidityDelta,
+			)
+
+			if tc.expectedError {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				fmt.Println("=======")
+				fmt.Println(actualAmount0.String())
+				fmt.Println(actualAmount1.String())
+				s.Require().Equal(actualAmount0, tc.amount0Expected)
+				s.Require().Equal(actualAmount1, tc.amount1Expected)
+
+				// validate if position has been properly updated
+				s.validatePositionUpdate(s.Ctx, tc.poolId, s.TestAccs[tc.ownerIndex], tc.lowerTick, tc.upperTick, tc.expectedPositionLiquidity)
```

Comment:
Let's also validate tick updates. There is a helper for that called `validateTickUpdates`

Reply:
done